src/classes/ObjectSetting.cls

```diff
@@ -0,0 +1,24 @@
+/**
+ * Created by User on 12.10.2017.
+ */
+
+public with sharing class ObjectSetting {
+    //fields
+    public String objectType { get; set; }
+    public String childObjectType { get; set; }
+    public String lookupField { get; set; }
+    public String title { get; set; }
+    public String icon { get; set; }
+    public List<String> fields = new List<String>{
+    };
+
+    //constructor
+    public ObjectSetting(String ot, String cot, String lf, String t, String i, List<String> f) {
```

Comment:
object setting is a wrapper around metadata object and should take as parameter __mdt object